### DISCUSSION TITLE
Is there anything specific in `poetry` for using this?

### DISCUSSION THREAD

**Discussion by JJ** [nat-n/poethepoet#104] [Ideas]
Long question is, is there anything specific in the way `poetry` organizes `pyproject.toml` or the virtual environments that prevents this from being used with the poetry alternative, [`hatch`](https://hatch.pypa.io/latest/environment/)?

**Comment by nat-n** [comment]
Hi @JJ, 

Nothing particularly special no.

I don't know much about hatch, but if it's basically like poetry then it should be possible to create another executor class like [the one for poetry](https://github.com/nat-n/poethepoet/blob/main/poethepoet/executor/poetry.py), and extend the [automatic executor selection logic here](https://github.com/nat-n/poethepoet/blob/main/poethepoet/executor/base.py#L93) and then poe should work with hatch just as well as poetry :)

See [the executor section of the readme](https://github.com/nat-n/poethepoet#change-the-executor-type) for more context.

I would in principle be open to contributions along these lines.
